### DISCUSSION TITLE
Question on Service Sale

### DISCUSSION THREAD

**Discussion by firdaussoberi** [massimo-zaniboni/asterisell-v5#12] [Q&A] [answered]
Hi, I got a quick question on the topics of service sale.
First of all I just want to thank the author for this nice work.

For our purpose, we are assigning a line rental when registering a new extension. This will be charged regardless whether the customer is using the call minutes or not, at per call basis on defined rate.
On the service sale, I saw it is mentioned that:
>A limitation of current version of Asterisell it is that in case there is no any call in the time frame, the services are not billed. But it is enough a call of a customer in a time-frame, for starting the billing of the services of all customers. So there can be problems only in case of a frozen Asterisell instance not processing any CDR.

So my question, is this also true currently in v6.4?
If so, we are thinking of generating a pseudo cdr every month, with zero cost (and then possibly hidden in invoice report) to counteract this. Or is there a better way in your opinion to do it?

Thanks.

p/s: Also, in we case want private help later (e.g as a service), where should we ask for it? Could we shoot an email?

**Comment by massimo-zaniboni** [comment, marked as answer]
> So my question, is this also true currently in v6.4?

Yes, it is still true, but in reality it is not a real limitation. Probably I wrote it, not in a clear way. If you have 100 customers, C1, C2, ... C100, with monthly billed service, it is sufficient that one customer has an inbound or outbound call in the month, for activating the calc of services also for all other customers. 

In practice, services are not calculated only if you have an Asterisell instance without any call, but used only for services.

Asterisell calculates automatically CDRS and services when there are changes, e.g. new calls, new/updated customers, new/updated rates, changes in configurations, and so on. If don't touch anything for a month, and there are no new calls, it thinks that there is nothing to update in the calcs of CDRS. If in a month there is no activating events, it does not update the calcs.

**Reply by firdaussoberi** [reply]
Hi thanks for the quick answer.

Somewhat related is the bundle rates as in:
https://www.asterisell.com/manual/bundle-rate-specification-language.html

We are also thinking of utilizing this as an alternative.
Could we do say something like this:

limit-on-first-seconds: 0

which means the bundle rate is still listed on invoice and the 1st second call is rated accordingly?

And, how would proportional rate works on nested organization hierarchy. Say if we have this:

limits-are-proportionals-to-activation-date: true

and hierarchy A/B and A/C where A is the billable party (where invoice is generated) and B and C are the two extensions.
B and C follows same bundle rate as A which is say $10, however B was registered for the full month and C was registered on the 15th (50%).

By design, would the total subscription/bundle rate displayed on the invoice to party A be $15 (10+5) then?
Would appreciate your response.
Thanks.

**Reply by massimo-zaniboni** [reply]
You should create another discussion with the title "Question on Bundle Rates".

**Reply by firdaussoberi** [reply]
Will do, thanks.

**Comment by massimo-zaniboni** [comment]
> p/s: Also, in we case want private help later (e.g as a service), where should we ask for it? Could we shoot an email?

yes, support@asterisell.com
